_posts/2021-08-31-going-zookeeperless-with-debezium-container-image-for-apache-kafka.adoc

```diff
@@ -0,0 +1,147 @@
+---
+layout: post
+title: Going ZooKeeper-less With the Debezium Container Image for Apache Kafka
+date: 2021-08-31
+tags: [ docker, apache kafka, examples ]
+author: gmorling
+---
+
+https://blogs.apache.org/kafka/entry/what-s-new-in-apache5[Apache Kafka 2.8] allows for a first glimpse into the ZooKeeper-less future of the widely used event streaming platform:
+shipping with a preview of https://cwiki.apache.org/confluence/display/KAFKA/KIP-500%3A+Replace+ZooKeeper+with+a+Self-Managed+Metadata+Quorum[KIP-500] ("Replace ZooKeeper with a Self-Managed Metadata Quorum"),
+you can now run Kafka clusters without the need for setting up and operating Apache ZooKeeper.
+This does not only simplify running Kafka from an operational perspective,
+the new metadata quorum implementation
+(named "KRaft", Kafka Raft metadata mode)
+also should provide much better scaling characteristics,
+for instance when it comes to large numbers of topics and partitions.
+
++++<!-- more -->+++
+
+https://www.morling.dev/blog/exploring-zookeeper-less-kafka/[I have blogged] about my first experiences with KRaft and ZooKeeper-less Kafka a while ago over on my personal blog.
+Since then, the Debezium community has upgraded its https://hub.docker.com/r/debezium/kafka[container image] for Apache Kafka to version 2.8,
+which makes it really simple to start your own explorations around running Kafka without ZooKeeper.
+Note that KRaft mode is an early access feature of Apache Kafka at this point in time and should not be used for production purposes.
+
+== Enabling KRaft Mode
+
+As of Debezium version 1.7, the container image for Kafka supports three new environment variables,
+to be specified when running the image:
+
+* `CLUSTER_ID`: A unique id, such as "5Yr1SIgYQz-b-dgRabWx4g"; if this variable is present, KRaft mode is enabled. Otherwise, the image behaves as before, and must be configured with a reference to ZooKeeper. When using KRaft mode, all nodes of the Kafka cluster must use the same cluster id
```

Comment:
maybe node that it needs to be generated with that `kafka-storage.sh` tool? I think this is still needed?

Reply:
Straight to the pain point, I like that :) I was weaseling around here a bit as I wasn't quite sure whether you _have_ to generate ids using that script or not. I'll explore and clarify.